doc/languages-frameworks/gnome.section.md

```diff
@@ -184,21 +184,26 @@ python3.pkgs.buildPythonApplication {
 And for a QT app like:
 
 ```nix
-mkDerivation {
+stdenv.mkDerivation {
   pname = "calibre";
   version = "3.47.0";
 
   nativeBuildInputs = [
     wrapGAppsHook3
+    qt5.wrapQtAppsHook
     qmake
     # ...
   ];
 
+  dontWrapQtApps = true;
   dontWrapGApps = true;
 
-  # Arguments to be passed to `makeWrapper`, only used by qt5’s mkDerivation
   preFixup = ''
-    qtWrapperArgs+=("''${gappsWrapperArgs[@]}")
+    for program in $out/bin/*; do
```

Comment:
Do we need explicit `wrapProgram`? This omits e.g. libexecdir. There are quite a few instances of the existing code in the repo:

https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+gappsWrapperArgs+qtWrapperArgs+&type=code&p=1

Reply:
I'm not sure. The `calibre` package does it this way, but I'm not sure if that's a reflection of best practice:

https://github.com/NixOS/nixpkgs/blob/61253596816c4cd65e2a0f474cbc0ac0c6e0f7cf/pkgs/applications/misc/calibre/default.nix#L181-L186

Reply:
I'll remove the `preFixup` phase and instead use something like this:

```nix
stdenv.mkDerivation {
  # ...
  makeWrapperArgs = [ "\${qtWrapperArgs[@]}" "\${gappsWrapperArgs[@]}"]; 
  # ...
};
```

Reply:
`makeWrapperArgs` is Python deriver thing, IIRC.

Reply:
Oh, hmm. Looks like this is the normal non-Python way:

https://github.com/NixOS/nixpkgs/blob/2e76e61c3cddad64ba6e805971e5334739ec6d46/pkgs/applications/kde/kasts.nix#L55-L58

Reply:
Modified in latest push to use something similar.